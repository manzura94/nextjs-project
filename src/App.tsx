import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'
import { ControlledFormPage } from './pages/ControlledFormPage'
import { Main } from './pages/Main'
import { UncontrolledFormPage } from './pages/UncontrolledFormPage'

function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Main />} />
        <Route path="/form-uncontrolled" element={<UncontrolledFormPage />} />
        <Route path="/form-controlled" element={<ControlledFormPage />} />
      </Routes>
    </Router>
  )
}

export default App
