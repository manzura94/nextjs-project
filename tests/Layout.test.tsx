import { render } from '@testing-library/react';
import React from 'react';
import App, { Layout } from '../app/root';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

vi.mock('@remix-run/react', () => ({
    Meta: () => <meta />,
    Links: () => <link />,
    ScrollRestoration: () => <div />,
    Scripts: () => <script />,

    Outlet: () => <div>Outlet Content</div>,
}));

vi.mock('./contexts/ThemeContext', () => ({
    ThemeProvider: ({ children }) => <div>{children}</div>,
}));

describe('Layout Component', () => {
    it('should render Meta, Links, and children correctly', () => {
        const { getByText, container } = render(
            <Layout>
                <div>Test Content</div>
            </Layout>
        );

        expect(getByText('Test Content')).toBeInTheDocument();
        expect(container.querySelector('meta')).toBeInTheDocument();
        expect(container.querySelector('link')).toBeInTheDocument();
        expect(container.querySelector('script')).toBeInTheDocument();
    });
});

describe('App Component', () => {
    it('should render Outlet inside ThemeProvider', () => {
        const { getByText, container } = render(
            <MemoryRouter>
                <App />
            </MemoryRouter>
        );

        expect(getByText('Outlet Content')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
