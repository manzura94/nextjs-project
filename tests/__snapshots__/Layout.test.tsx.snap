// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > should render Outlet inside ThemeProvider 1`] = `
<div>
  <div>
    Outlet Content
  </div>
</div>
`;
